gridstatus/ercot.py

```diff
@@ -2029,37 +2024,231 @@ def _handle_sced_timestamp(self, df, verbose=False):
+        elif end is None:
+            if shadow_prices:
+                published_after = date.normalize()
+                published_before = published_after + pd.DateOffset(days=1)
+            else:
+                friendly_name_timestamp_after = date.normalize()
+                friendly_name_timestamp_before = (
+                    friendly_name_timestamp_after + pd.DateOffset(days=1)
+                )
+
+            # date needs to be set to none so the docs are not filtered on the date
+            date = None
         else:
-            df = pd.concat(all_dfs)
 
-        df = self._handle_sced_timestamp(df, verbose=verbose)
+            if shadow_prices:
+                published_after = date
+                published_before = end
 
-        df["SystemLambda"] = df["SystemLambda"].astype("float64")
+            else:
+                friendly_name_timestamp_after = date
+                friendly_name_timestamp_before = end
 
-        df = df.rename(
-            columns={
-                "SystemLambda": "System Lambda",
-            },
+            date = None
+
+        docs = self._get_documents(
+            report_type_id=report_type_id,
+            date=date,
+            published_after=published_after,
+            published_before=published_before,
+            friendly_name_timestamp_after=friendly_name_timestamp_after,
+            friendly_name_timestamp_before=friendly_name_timestamp_before,
+            verbose=verbose,
+            constructed_name_contains="csv.zip",
+        )
+
+        return docs
+
+    def _handle_sced_docs(
+        self,
+        docs,
+        verbose,
+        desc,
+        column_name_mapper,
+        front_columns,
+        read_csv_kwargs={},
+    ):
+        """
+        Handles a list of SCED documents
+
+        Args:
+            docs (list): list of Document
+            verbose (bool): print verbose output
+            desc (str): description for tqdm
+            column_name_mapper (dict): dictionary to map column names
+            front_columns (list): columns to move to the front
+            read_csv_kwargs (dict): kwargs to pass to pd.read_csv
+
+        Returns:
+            pandas.DataFrame: A DataFrame with SCED data sorted by SCED Timestamp
+
+        """
+        all_dfs = []
+
+        for doc in tqdm.tqdm(docs, desc=desc, disable=not verbose):
+            log(f"Reading {doc.url}", verbose)
+            df = pd.read_csv(doc.url, compression="zip", **read_csv_kwargs)
+            df["Publish Time"] = pd.to_datetime(doc.publish_date)
+            all_dfs.append(df)
+
+        df = pd.concat(all_dfs)
+        df = self._handle_sced_timestamp(df, verbose=verbose)
+        df = df.rename(columns=column_name_mapper)
+
+        if front_columns:
+            df = utils.move_cols_to_front(df, front_columns)
+
+        return df.sort_values("SCED Timestamp").reset_index(drop=True)
+
+    def _construct_limiting_facility_column(self, df):
+        df["Limiting Facility"] = (
```

Comment:
"Limiting Facility" added at the request of @cwaldoch.

Reply:
if contigency is base case, this should be null.